server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymRuleAction.java

```diff
@@ -0,0 +1,129 @@
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class GetSynonymRuleAction extends ActionType<GetSynonymRuleAction.Response> {
+
+    public static final GetSynonymRuleAction INSTANCE = new GetSynonymRuleAction();
+    public static final String NAME = "cluster:admin/synonym_rules/get";
+
+    public GetSynonymRuleAction() {
+        super(NAME, GetSynonymRuleAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+        private final String synonymsSetId;
+
+        private final String synonymRuleId;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.synonymsSetId = in.readString();
+            this.synonymRuleId = in.readString();
+        }
+
+        public Request(String synonymsSetId, String synonymRuleId) {
+            this.synonymsSetId = synonymsSetId;
+            this.synonymRuleId = synonymRuleId;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+            if (Strings.isNullOrEmpty(synonymsSetId)) {
+                validationException = ValidateActions.addValidationError("synonyms set must be specified", validationException);
+            }
+            if (Strings.isNullOrEmpty(synonymRuleId)) {
+                validationException = ValidateActions.addValidationError("synonym rule id must be specified", validationException);
+            }
+
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(synonymsSetId);
+            out.writeString(synonymRuleId);
+        }
+
+        public String synonymsSetId() {
+            return synonymsSetId;
+        }
+
+        public String synonymRuleId() {
+            return synonymRuleId;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request request = (Request) o;
+            return Objects.equals(synonymsSetId, request.synonymsSetId) && Objects.equals(synonymRuleId, request.synonymRuleId);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(synonymsSetId, synonymRuleId);
+        }
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private final SynonymRule synonymRule;
+
+        public Response(StreamInput in) throws IOException {
+            this.synonymRule = new SynonymRule(in);
+        }
+
+        public Response(SynonymRule synonymRule) {
+            this.synonymRule = synonymRule;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            synonymRule.writeTo(out);
```

Comment:
Do we need to call `super.writeTo(out)` here?

Reply:
Thanks for checking! We can't, it's an abstract method for the super class.

Reply:
Yes, that's for `Request`, not `Response`. They have different base classes 👍 . The call for super.writeTo(out) is present on the Request 👍